cmd/libs/go2idl/make-client/output/deployment.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file was autogenerated by the command:
+// $ /tmp/go-build983536827/command-line-arguments/_obj/exe/main
+// Do not edit it manually!
+
+package unversionedClient
+
+import (
+	api "k8s.io/kubernetes/pkg/api"
+	extensions "k8s.io/kubernetes/pkg/apis/extensions"
+	fields "k8s.io/kubernetes/pkg/fields"
+	labels "k8s.io/kubernetes/pkg/labels"
+	watch "k8s.io/kubernetes/pkg/watch"
+)
+
+// DeploymentNamespacer has methods to work with Deployment resources in a namespace
+type DeploymentNamespacer interface {
+	Deployment(namespace string) DeploymentInterface
+}
+
+// DeploymentInterface has methods to work with Deployment resources.
+type DeploymentInterface interface {
+	List(label labels.Selector, field fields.Selector) (*extensions.DeploymentList, error)
+	Get(name string) (*extensions.Deployment, error)
+	Delete(name string, options *api.DeleteOptions) error
+	Create(*extensions.Deployment) (*extensions.Deployment, error)
+	Update(*extensions.Deployment) (*extensions.Deployment, error)
+	UpdateStatus(*extensions.Deployment) (*extensions.Deployment, error)
+	Watch(label labels.Selector, field fields.Selector, opts api.ListOptions) (watch.Interface, error)
+}
+
+// deployments implements DeploymentInterface
+type deployments struct {
+	client *ExtensionsClient
+	ns     string
+}
+
+// newDeployments returns a Deployments
+func newDeployments(c *ExtensionsClient, namespace string) *deployments {
```

Comment:
I agree with @krousey that it should be RESTClient.
But I support @caesarxuchao here - we should start with replicating what we have, and then it will be so much easier to change it.
So I would leave it as is for now (possibly adding a TODO to fix it).